src/sagemaker_xgboost_container/algorithm_mode/train_utils.py

```diff
@@ -33,7 +33,9 @@ def get_union_metrics(metric_a, metric_b):
     elif metric_b is None:
         return metric_a
     else:
-        metric_list = list(set(metric_a).union(metric_b))
+        # The order of metric_list need to be consistent among all hosts in distributed training
+        # So we have metric_list sorted here.
+        metric_list = sorted(list(set(metric_a).union(metric_b)))
```

Comment:
Not sure if it matters, but this is returning a generator, which means it can be exhausted unlike a list. I would recommend one of the following two changes:

```suggestion
        sorted_union_metric_list = list(sorted(list(set(metric_a).union(metric_b))))
        return sorted_union_metric_list
```

or

```suggestion
        union_metric_list = metric_a + [metric for metric in metric_b if metric not in metric_a]
        return union_metric_list
```

Reply:
Hi, Thanks for the comment, by this doc : https://docs.python.org/3/library/functions.html#sorted sorted() return a list. And I also try to print(type(sorted(metric_list))) and got a list type. 
And for union_metric_list = metric_a + [metric for metric in metric_b if metric not in metric_a] I think set's union method has a better time complexity as they are using hash values.

Reply:
Ah, you're right - my bad! LGTM then.